web/viewer.js

```diff
@@ -1061,9 +1062,13 @@ var PDFViewerApplication = {
     });
   },
 
-  setInitialView: function pdfViewSetInitialView(storedHash, scale) {
+  setInitialView: function pdfViewSetInitialView(storedHash, scale, rotation) {
     this.isInitialViewSet = true;
 
+    if (rotation !== this.pageRotation) {
+      this.rotatePages(rotation);
```

Comment:
I get `pdfViewSetScale: 'null' is an unknown zoom value` in the console when I run the preview. I think the problem is that `setInitialView` is called from multiple places and not all of them provide the `rotation` argument, thus it is `null` in those cases. This `if` statement does not check this and thus `this.rotatePages` receives `null` as an argument. Therefore you might need to change the `if` statement to `if (rotation && rotation !== this.pageRotation) {`.